src/doc/src/reference/features.md

```diff
@@ -513,3 +513,9 @@ source and inspect it.
 
 [`cargo vendor`]: ../commands/cargo-vendor.md
 [cargo-clone-crate]: https://crates.io/crates/cargo-clone-crate
+
+### Caveats
+
+Features, which are a form of conditional compilation, require an exponential number of configurations and test cases to be 100% covered. By default, testing, linting, Miri, docs and others will only run on the default set of features.
+
+There are certain tools like [cargo-hack](https://github.com/taiki-e/cargo-hack) and [cargo-all-features](https://github.com/frewsxcv/cargo-all-features) that allow you to test all combinations of features which can help reduce the amount of configuraton needed.
```

Comment:
I don't think we can make suggestions of third party tools to use here.  

Instead, perhaps it can just encourage the reader to consider their strategy for how they are going to test the different configurations. They may want to consider testing no default features, with default features, all features, or any combination of features.  They'll need to consider their resource and time budget for how to approach it.

Reply:
Changed the wording, does this work?